src/algebraic_geometry/prime_spectrum.lean

```diff
@@ -421,6 +432,24 @@ def basic_open (r : R) : topological_space.opens (prime_spectrum R) :=
 lemma is_open_basic_open {a : R} : is_open ((basic_open a) : set (prime_spectrum R)) :=
 (basic_open a).property
 
+lemma basic_open_eq_zero_locus_compl (r : R) :
+  (basic_open r : set (prime_spectrum R)) = (zero_locus {r})ᶜ :=
+set.ext $ λ x, by simpa only [set.mem_compl_eq, mem_zero_locus, set.singleton_subset_iff]
+
+lemma is_topological_basis_basic_opens :
+  topological_space.is_topological_basis (set.range (λ r, (@basic_open R _ r).val)) :=
```

Comment:
Another minor comment:
Just for the sake of consistency with `basic_open_eq_zero_locus_compl`, it might look nicer to change `set.range (λ r, (@basic_open R _ r).val)` to `set.range (λ (r : R), (basic_open r : set (prime_spectrum R)))`.

Reply:
Good point!